File: app/src/main/java/com/returnnotfound/stain/utils/DimensionUtils.java
package com.returnnotfound.stain.utils;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;

public abstract class DimensionUtils {
  public static int dpToPx(Context context, float dp) {
    return (int) (dp * ((float) context.getResources().getDisplayMetrics().densityDpi / DisplayMetrics.DENSITY_DEFAULT));
  }

  public static float pxToDp(Context context, float px) {
    if (context == null || context.getResources() == null) {
      return 0;
    }
    return px / ((float) context.getResources().getDisplayMetrics().densityDpi / DisplayMetrics.DENSITY_DEFAULT);
  }

  public static float getWidthScreenDp(Context context) {
    DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
    return displayMetrics.widthPixels / displayMetrics.density;
  }

  public static float getHeightScreenDp(Context context) {
    DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
    return displayMetrics.heightPixels / displayMetrics.density;
  }

  public static int getStatusBarHeight(Context context) {
    int result = 0;
    int resourceId = context.getResources().getIdentifier("status_bar_height", "dimen", "android");
    if (resourceId > 0) {
      result = context.getResources().getDimensionPixelSize(resourceId);
    }
    return result;
  }

  public static int getActionBarHeight(Context context) {
    TypedValue tv = new TypedValue();
    if (context.getTheme().resolveAttribute(android.R.attr.actionBarSize, tv, true)) {
      return TypedValue.complexToDimensionPixelSize(tv.data, context.getResources().getDisplayMetrics());
    }
    return 0;
  }
}

File: app/src/main/java/com/returnnotfound/stain/base/utils/DimensionUtils.kt
package com.returnnotfound.stain.base.utils

import android.R
import android.content.Context
import android.util.DisplayMetrics
import android.util.TypedValue

object DimensionUtils {

  @JvmStatic
  fun dpToPx(context: Context, dp: Float): Int {
    return (dp * (context.resources
      .displayMetrics.densityDpi.toFloat() / DisplayMetrics.DENSITY_DEFAULT)).toInt()
  }

  @JvmStatic
  fun pxToDp(context: Context?, px: Float): Float {
    return if (context == null || context.resources == null) 0F
    else px / (context.resources.displayMetrics.densityDpi.toFloat() / DisplayMetrics.DENSITY_DEFAULT)
  }

  @JvmStatic
  fun getWidthScreenDp(context: Context): Float {
    val displayMetrics = context.resources.displayMetrics
    return displayMetrics.widthPixels / displayMetrics.density
  }

  @JvmStatic
  fun getHeightScreenDp(context: Context): Float {
    val displayMetrics = context.resources.displayMetrics
    return displayMetrics.heightPixels / displayMetrics.density
  }

  @JvmStatic
  fun getStatusBarHeight(context: Context): Int {
    var result = 0
    val resourceId =
      context.resources.getIdentifier("status_bar_height", "dimen", "android")
    if (resourceId > 0) {
      result = context.resources.getDimensionPixelSize(resourceId)
    }
    return result
  }

  @JvmStatic
  fun getActionBarHeight(context: Context): Int {
    val tv = TypedValue()
    return if (context.theme.resolveAttribute(R.attr.actionBarSize, tv, true))
      TypedValue.complexToDimensionPixelSize(
        tv.data, context.resources.displayMetrics
      )
    else 0
  }
}